testing/sdk_repository.py

```diff
@@ -96,18 +100,194 @@ def remove_universe_repos(stub_urls):
     log.info('Finished removing universe repos')
 
 
-def universe_session():
+def add_package_registry_stub() -> dict:
+    stub_url = os.environ['PACKAGE_REGISTRY_STUB_URL']
+    with urllib.request.urlopen(stub_url) as url:
+        repo = json.loads(url.read().decode())
+        min_supported = repo["packages"][0]["minDcosReleaseVersion"]
+
+    version = sdk_cmd.dcos_version()
+
+    if not is_repo_supported(version, min_supported):
+        raise Exception('DC/OS {} does not support package registry. '
+                        'Minimum required {}'.format(version, min_supported))
+    return add_stub_universe_urls([stub_url])
+
+
+def install_package_registry(service_secret_path: str) -> dict:
+    # Install Package Registry
+    sdk_install.install(
+        PACKAGE_REGISTRY_NAME,
+        PACKAGE_REGISTRY_SERVICE_NAME,
+        expected_running_tasks=0,
+        additional_options={
+            'registry': {
+                'service-account-secret-path': service_secret_path
+            }
+        },
+        wait_for_deployment=False,
+        insert_strict_options=False
+    )
+
+    pkg_reg_repo = add_stub_universe_urls(
+        ['https://registry.marathon.l4lb.thisdcos.directory/repo']
+    )
+    # If `describe` endpoint is working, registry is writable by AR.
+    package_name = 'hello'
+    package_version = 'world'
+    expected_msg = 'Version [{}] of package [{}] ' \
+                   'not found'.format(package_version, package_name)
+    loop_until_cli_condition(
+        'registry describe --package-name={} --package-version={}'.format(
+            package_name, package_version
+        ),
+        lambda code, out, err: code == 1 and expected_msg in err
+    )
+    return pkg_reg_repo
+
+
+def add_dcos_files_to_registry(
+        temp_dir
+) -> None:
+    # Use DCOS_FILES_PATH if its set to a valid path OR use pytest's tmpdir.
+    dcos_files_path = os.environ.get('DCOS_FILES_PATH', '')
+    if not os.path.isdir(dcos_files_path):
+        dcos_files_path = str(temp_dir)
+    stub_universe_urls = get_universe_repos()
+    log.info('Using {} to build .dcos files (if not exists) from {}'.format(
+        dcos_files_path, stub_universe_urls
+    ))
+    dcos_files_list = build_dcos_files_from_stubs(
+        stub_universe_urls,
+        dcos_files_path,
+        temp_dir
+    )
+    log.info('Bundled .dcos files : {}'.format(dcos_files_list))
+    for file in dcos_files_list:
+        rc, out, err = sdk_cmd.run_raw_cli(' '.join([
+            'registry',
+            'add',
+            '--dcos-file={}'.format(file),
+            '--json'
+        ]))
+        assert rc == 0
+        assert len(json.loads(out)['packages']) > 0
+
+
+def build_dcos_files_from_stubs(
+        stub_universe_urls: list,
+        dcos_files_path: str,
+        temp_dir
+) -> list:
+    if not len(stub_universe_urls):
+        return stub_universe_urls
+    package_file_paths = []
+    for repo_url in stub_universe_urls:
+        headers = {
```

Comment:
Is this not also supposed to handle the authentication with the DC/OS service? We have some other "request" functions including `service_request` and `cluster_request` in `sdk_cmd` which may be doing similar things. If would be good to understand what is being done differently here.

It may also make sense to group these types of functions more logically, although that is probably out of scope for this PR.

Reply:
No, the urls are supposed to be publicly accessible (at least as of today). An example url would be `https://universe-converter.mesosphere.com/transform?url=https://infinity-artifacts.s3.amazonaws.com/autodelete7d/template/20180410-181638-EhC0WXM14amJN6c7/stub-universe-template.json`